test/tables_tests.jl

```diff
@@ -0,0 +1,134 @@
+            @test_throws Exception Tables.getcolumn(chn, :i)
+            @test_throws Exception Tables.getcolumn(chn, 11)
+            @test Tables.rowaccess(typeof(chn))
+            @test Tables.rows(chn) === chn
+            @test length(Tables.rowtable(chn)) == 4000
+            nt = Tables.rowtable(chn)[1]
+            @test nt ==
+                  (; (k => Tables.getcolumn(chn, k)[1] for k in Tables.columnnames(chn))...)
+            @test nt == collect(Iterators.take(Tables.namedtupleiterator(chn), 1))[1]
+            nt = Tables.rowtable(chn)[2]
+            @test nt ==
+                  (; (k => Tables.getcolumn(chn, k)[2] for k in Tables.columnnames(chn))...)
+            @test nt == collect(Iterators.take(Tables.namedtupleiterator(chn), 2))[2]
+            @test Tables.schema(chn) isa Tables.Schema
+            @test Tables.schema(chn).names ===
+                  (:Iteration, :Chain, :a, :b, :c, :d, :e, :f, :g, :h)
+            @test Tables.schema(chn).types === (
+                Int64,
+                Int64,
+                Float64,
+                Float64,
+                Float64,
+                Float64,
+                Float64,
+                Float64,
+                Float64,
+                Float64,
+            )
+            @test Tables.matrix(chn[:, :, 1])[:, 3:end] ≈ chn[:, :, 1].value
+            @test Tables.matrix(chn[:, :, 2])[:, 3:end] ≈ chn[:, :, 2].value
+        end
+
+        @testset "TableTraits interface" begin
+            @test IteratorInterfaceExtensions.isiterable(chn)
+            @test TableTraits.isiterabletable(chn)
+            nt = collect(Iterators.take(IteratorInterfaceExtensions.getiterator(chn), 1))[1]
+            @test nt ==
+                  (; (k => Tables.getcolumn(chn, k)[1] for k in Tables.columnnames(chn))...)
+            nt = collect(Iterators.take(IteratorInterfaceExtensions.getiterator(chn), 2))[2]
+            @test nt ==
+                  (; (k => Tables.getcolumn(chn, k)[2] for k in Tables.columnnames(chn))...)
+        end
+
+        @testset "DataFrames.DataFrame constructor" begin
+            @inferred DataFrame(chn)
+            @test DataFrame(chn) isa DataFrame
+            df = DataFrame(chn)
+            @test Tables.columntable(df) == Tables.columntable(chn)
+        end
+    end
+
+    @testset "ChainDataFrames" begin
+        val = rand(1000, 8, 4)
+        colnames = ["a", "b", "c", "d", "e", "f", "g", "h"]
+        internal_colnames = ["c", "d", "e", "f", "g", "h"]
+        chn = Chains(val, colnames, Dict(:internals => internal_colnames))
+        cdf = describe(chn)[1]
+
+        @testset "Tables interface" begin
+            @test Tables.istable(typeof(cdf))
+            @test Tables.columnaccess(typeof(cdf))
+            @test Tables.columns(cdf) === cdf
+            @test Tables.columnnames(cdf) == keys(cdf.nt)
+            @testset for k in keys(cdf.nt)
+                @test Tables.getcolumn(cdf, k) == getproperty(cdf.nt, k)
```

Comment:
I'd suggest avoiding `getproperty` in the tests as well:
```suggestion
            for (k, v) in pairs(cdf.nt)
                @test Tables.getcolumn(cdf, k) == v
```